src/main/java/sg/ncl/MainController.java

```diff
@@ -3341,6 +3347,162 @@ public String adminEnergy(Model model,
+        if (reservationStatusForm.getNumNodes() == null) {
+            // number of nodes not fill
+            // release all nodes; endpoint is the same with adminNodesReservation but with different HTTP method
+            response = restTemplate.exchange(properties.getReservationStatus(reservationStatusForm.getTeamId()), HttpMethod.DELETE, request, String.class);
+        } else {
+            // release a specific number of nodes
+            response = restTemplate.exchange((properties.releaseNodes(reservationStatusForm.getTeamId(), reservationStatusForm.getNumNodes())), HttpMethod.DELETE, request, String.class);
+        }
+
+        JSONObject object = new JSONObject(response.getBody().toString());
+        String status = object.getString(STATUS);
+        String msg = object.optString(MESSAGE);
+
+        if ("release reservation OK".equals(status) && (msg.equals("[]"))) {
+            msg = "No nodes to be released";
+        } else if ("release reservation OK".equals(status)) {
+            // node ids
+            // ["A","B"] -> A,B
+            msg = msg.replace("[", "").replace("]", "").replace("\"", "");
+        }
+
+        redirectAttributes.addFlashAttribute(STATUS, status);
+        redirectAttributes.addFlashAttribute(MESSAGE, msg);
+    }
+
+    private void reserveNodes(ReservationStatusForm reservationStatusForm, RedirectAttributes redirectAttributes) {
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+        ResponseEntity response = null;
+
+        if (reservationStatusForm.getNumNodes() == null) {
+            redirectAttributes.addFlashAttribute(STATUS, NODES_RESERVATION_FAIL);
+            redirectAttributes.addFlashAttribute(MESSAGE, "Number of nodes not specify");
+            return;
+        }
+
+        if (reservationStatusForm.getMachineType() == null) {
+            // machine type not filled
+            // endpoint is the same as releaseNodes, i.e. /{id}
+            response = restTemplate.exchange(properties.reserveNodes(reservationStatusForm.getTeamId(), reservationStatusForm.getNumNodes(), null), HttpMethod.POST, request, String.class);
+        } else {
+            // reserve a specific number of machine type
+            response = restTemplate.exchange((properties.reserveNodes(reservationStatusForm.getTeamId(), reservationStatusForm.getNumNodes(), reservationStatusForm.getMachineType())), HttpMethod.POST, request, String.class);
+        }
+
+        JSONObject object = new JSONObject(response.getBody().toString());
+        String status = object.getString(STATUS);
+        String msg = object.optString(MESSAGE);
+
+        if ("nodes reservation OK".equals(status) && (msg.equals("[]"))) {
+            msg = "No nodes to be reserved";
+        } else if ("nodes reservation OK".equals(status)) {
+            // node ids
+            // ["A","B"] -> A,B
+            msg = msg.replace("[", "").replace("]", "").replace("\"", "");
+        }
+
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)